go/interpreter/lfvm/instructions_test.go

```diff
@@ -1796,6 +1796,61 @@ func TestInstructions_ComparisonAndShiftOperations(t *testing.T) {
 	}
 }
 
+func TestInstructions_OpExtCodeCopy_CallsContextAndCopiesCodeSlice(t *testing.T) {
+
+	code := []byte{0x1, 0x2, 0x3, 0x4}
+	address := tosca.Address{}
+	_, _ = rand.Read(address[:])
+	var offset uint64 = 1
+	var size uint64 = 2
+
+	runContext := tosca.NewMockRunContext(gomock.NewController(t))
+	runContext.EXPECT().GetCode(address).Return(code)
+
+	ctxt := getEmptyContext()
+	ctxt.context = runContext
+	ctxt.gas = 1 << 32
+
+	ctxt.stack = fillStack(
+		*new(uint256.Int).SetBytes(address[:]),
+		*uint256.NewInt(0),      // memOffset
+		*uint256.NewInt(offset), // codeOffset
+		*uint256.NewInt(size),   // length
+	)
+
+	err := opExtCodeCopy(&ctxt)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if want, got := code[offset:offset+size], ctxt.memory.store[0:size]; !bytes.Equal(want, got) {
+		t.Errorf("unexpected memory, wanted %v, got %v", want, got)
+	}
+}
+
+func TestInstructions_opExtcodesize_CallsContextAndWritesResultInStack(t *testing.T) {
+
+	address := tosca.Address{}
+	_, _ = rand.Read(address[:])
+	runContext := tosca.NewMockRunContext(gomock.NewController(t))
+	runContext.EXPECT().GetCodeSize(address).Return(1234)
+
+	ctxt := getEmptyContext()
+	ctxt.context = runContext
+	ctxt.gas = 1 << 32
```

Comment:
same here, you don't need that change. It can confusing because `getEmptyContext` does not generate an empty context but a prefilled one, with more than enough gas.

Reply:
maybe we should rename empty context to test context. I always thought an empty context would have no gas.